### DISCUSSION TITLE
Interpretation of results

### DISCUSSION THREAD

**Discussion by LauraMarciano91** [cdriveraus/ctsem#16] [Q&A]
Hi Charles,
I have three questions for you. 
The first is about the interpretation of the results. As I understood, in the output of a continuous model I can compute discrete values of the drift matrix according to a particular time interval. So, in the following outcome, I ask for a timeinterval=2 (which should be 2 days). Since the values of the continuous drift matrix are negative (but they are for the whole period I assume), but the dtDRIFT is only positive, which is the best way to interpret them? Can I assume that, although for the whole period e.g. Variable 1 shows a decrease, for two days it shows an increase as reported in the dtDRIFT? 
                      mean     sd    2.5%     50%   97.5%
drift_eta1      -0.1329 0.0446 -0.2400 -0.1262 -0.0659
drift_eta1_eta2  0.2374 0.0921  0.0653  0.2364  0.4128
drift_eta2_eta1  1.1033 0.1765  0.7645  1.1038  1.4611
drift_eta2      -4.7345 0.6877 -6.1443 -4.7323 -3.3749

 matrix row col    Mean     sd    2.5%     50%   97.5%
dtDRIFT   1   1  0.8500 0.0561  0.7203  0.8568  0.9418
dtDRIFT   1   2  0.0429 0.0155  0.0142  0.0432  0.0726
dtDRIFT   2   1  0.2057 0.0470  0.1219  0.2024  0.3048
dtDRIFT   2   2  0.0106 0.0050  0.0031  0.0099  0.0225

Additionally, how can I retrieve single subject values from an rstan model? I found the code e.g. DRIFT [8,2,1] to obtain the 2nd row and the 1st column of the DRIFT matrix for subject 8. I attached it to the summary but the program returns NULL.  Can you please share the precise R code to obtain this? 

Finally, when you suggested to use the following code:
if(FALSE){ #set to TRUE if you want to use it obviously...
  model_tot$pars$indvarying <- TRUE 
  model_tot$pars$sdscale[!model_tot$pars$matrix %in% c('T0MEANS','MANIFESTMEANS','CINT')] <- .05 #allow variation but regularize non-intercept variation
}
can you please explain better what it is doing and why I should "consider computing the mean of each subject and using it as a covariate (time independent predictor) to predict all parameters, and disable all the random effects"? I would like to compare models with different constraints but I am not sure how to do it.

In general, I am quite familiar with CLPM and RICLPM output and I am trying to understand here how to use a similar approach (like model comparison) and interpret the output in a similar way (using discrete parameters for easier interpretation, and also distinguishing between from within effects). 

Many thanks, LM

**Comment by cdriveraus** [comment]
Re interpretation, it sounds like you'd benefit from another careful look at one of the papers, as negative diagonals of the continuous time drift matrix are expected, since they describe how the system is *changing* -- higher values result in faster downwards change, lower values result in faster upwards change, so the system likely has an equilibrium / is dissipative. The discrete time parameters instead describe how the system looks after it has *changed* for some specific period of time -- higher values are still high but less so, lower values are still low but less so. Of my works the hierarchical Bayesian paper probably covers the continuous to discrete translation the best.

Individual subject parameters are not output by default because this was causing performance issues in many cases. To get them, you need to use the ctExtract function and specify that you want the individual parameters. 

The random effects / covariate issue is going to be the trickiest I guess -- basically you can estimate a big hierarchical model with everything varying across individuals, but this can be very slow depending on model / data. Setting a narrow prior for the random effects scale is one way to speed this up. Alternatively to random effects, using the observed mean as a covariate for the other parameters is a computationally cheaper way of getting at what is usually a substantial portion of the heterogeneity.

**Reply by lindyswilliams** [reply]
Hey Charles!
I had a quick question about the subject parameters. I was able to successfully use the ctExtract function to get samples for the subject parameters, but am I correct in my interpretation that these will be draws from the posterior distributions? I was hoping to retrieve the chains for the subject parameters. Is this possible?
Thank you!

**Reply by cdriveraus** [reply]
You are fitting with optimize=F then? Otherwise there are no chains and the posterior is an approximation based on the point estimates and uncertainty. Assuming you've waited for the full sampling (optimize=F), then ctExtract returns the post warmup draws, but you'll find the full stanfit object under fit$stanfit$stanfit (yes this is a poor choice of structure naming due to on-the-fly development planning), and the usual rstan tools can be applied to this.

**Reply by lindyswilliams** [reply]
Yes, I was fitting with optimize=F. Perfect, thank you so much!

**Comment by LauraMarciano91** [comment]
Many thanks for your answer. I noticed that dt parameters for the drift matrix are only computed when covariates are included in the model. In any case, the dt drift effects are already standardized - i.e. the interpretation at the end would be like auto and cross-lagged of a RI-CLPM? 

Additionally, using the narrow prior resulted in very restrict95%CI, thus also graphically lines show little variation. Is this correct? or better to remove this prior and allow for larger variation in estimates and better graphs? 
I pasted here the image I
 obtained for one predictor:

![image](https://user-images.githubusercontent.com/85165113/120649313-e6142200-c47c-11eb-9322-b9baa864d215.png)

**Reply by LauraMarciano91** [reply]
additionally, is it correct to interpret t0means, drift, and diff parameters in terms of within-person effects, whereas manifest means and manifestvar in terms of between-person effect? I am struggling a bit with the interpretation here.. can you maybe try to summarize it or provide an example in a way that is similar to a RI-CLPM for easier comprehension? Many thanks
